src/wallet/rpcwallet.cpp

```diff
@@ -4473,15 +4473,17 @@ static RPCHelpMan upgradewallet()
     if (!request.params[0].isNull()) {
         version = request.params[0].get_int();
     }
-
     bilingual_str error;
-    std::vector<bilingual_str> warnings;
-    if (!pwallet->UpgradeWallet(version, error, warnings)) {
+    if (!pwallet->UpgradeWallet(version, error)) {
         throw JSONRPCError(RPC_WALLET_ERROR, error.original);
     }
     UniValue obj(UniValue::VOBJ);
     if (!error.empty()) {
         obj.pushKV("error", error.original);
+    } else {
+        if(!request.params[0].isNull()) {
+            obj.pushKV("result", "The wallet was upgraded successfully to version 169900.");
+        }
```

Comment:
I think this addition can be dropped along with the test change (see also #20403).

Reply:
okay so I'll just take the result object out